### PR TITLE
pathlib training.py

### PR DESCRIPTION
**Pull request by cakester**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Phased refactor adoption of loader/segmentation_pair.py as detailed in #872 using pathlib for using pathlib for using pathlib for training.py

## Linked issues

Step wise remediation of #872

### CODE REVIEW

File: ivadomed/training.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
                 torch.save(state, resume_path)
```
Comment (dyt811) [inline ivadomed/training.py:293]: Could you double check if `torch.save` supports pathlib objects references? 
Its signature seems to suggest more `str` or file object? 

image

File: ivadomed/training.py [hunk 2]
```diff
@@ -310,39 +310,39 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
...
 
     # Save final model
-    final_model_path = os.path.join(path_output, "final_model.pt")
+    final_model_path = Path(path_output, "final_model.pt")
     torch.save(model, final_model_path)
 
     # Save best model in output path
-    if os.path.isfile(resume_path):
+    if resume_path.is_file():
         state = torch.load(resume_path)
```
Comment (dyt811) [inline ivadomed/training.py:318]: Above

**Review by dyt811 — COMMENTED** [review]
Thanks. Could you double check those `torch.save` and `torch.load` to ensure they work with pathlib or change to str.

**Comment by cakester** [comment]
this function _is_path is used by torch, note it checks isinstance(name_or_buffer, pathlib.Path)
image

**Review by dyt811 — APPROVED** [review]
Thanks for validating those details. All good. Thanks.

**Pull request merged**